File: BackEnd/apiFunctions/purchasing/item/document.php
<?php
//*************************************************************************************************
// FileName : document.php
// FilePath : apiFunctions/purchasing/item
// Author   : Christian Marty
// Date		: 01.08.2020
// License  : MIT
// Website  : www.christian-marty.ch
//*************************************************************************************************
declare(strict_types=1);
global $database;
global $api;

require_once __DIR__ . "/../../document/_document.php";

if($api->isGet(\Permission::PurchaseOrder_View))
{
	$parameters = $api->getGetData();
	if(!isset($parameters->PurchaseOrderNumber))$api->returnParameterMissingError('PurchaseOrderNumber');
	$purchaseOrderNumber = \Numbering\parser(\Numbering\Category::PurchaseOrder, $parameters->PurchaseOrderNumber);
	if(!$purchaseOrderNumber) $api->returnParameterError('PurchaseOrderNumber');

	$query = "SELECT DocumentIds FROM purchaseOrder WHERE PurchaseOrderNumber = '$purchaseOrderNumber';";
	$result = $database->query($query)[0]->DocumentIds??null;

	$api->returnData(\Document\getDocumentsFromIds($result));
}

File: API/document.py
# *************************************************************************************************
# FileName : document.py
# Author   : Christian Marty
# Date     : 16.02.2024
# License  : MIT
# Website  : www.christian-marty.ch
# *************************************************************************************************

class Document(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance
        self.ingest = self.Ingest(self)

    def list(self):
        data = self._eerp.get('document')
        return data

    def types(self):
        data = self._eerp.get('document/type')
        return data

    def item(self, document_code: str | int):
        data = self._eerp.get('document/item', {"DocumentNumber": document_code})
        return data

    class Ingest(object):
        def __init__(self, document_instance):
            self._document = document_instance

        def list(self):
            data = self._document.get('document/ingest/list')
            return data
